Fix incorrect z-order of dashboard map markers (fixes #6639)

aNsHuL5217 opened the pull request:
### Description
Fixes #6639

The dashboard map markers were previously rendered in the order they appeared in the DOM (which matches the text list order). This caused visual z-order issues where "Northern" markers (background) were sometimes drawn on top of "Southern" markers (foreground) if they appeared later in the list.

This PR fixes the issue by:
1. Collecting all user data from the DOM first.
2. Sorting the users by latitude in descending order (North to South).
3. Adding the markers to the map in this sorted order.

This ensures that Northern markers are always rendered first (in the background) and Southern markers are rendered last (in the foreground), preserving the correct visual perspective.

**Screenshot (After Fix):**
<img width="1023" height="557" alt="image (2)" src="https://github.com/user-attachments/assets/0673fa49-bab9-4bea-bfb7-c3c9c2f28a98" />

### How has this been tested?
I tested this locally in the development environment:
1. Created two users with overlapping coordinates using the Rails console:
   * **User A (Friend):** Latitude `51.501` (North)
   * **User B (Me):** Latitude `51.5` (South)
2. Verified on the Dashboard map that **User B (South)** is correctly drawn **in front of** User A (North).

hlfan commented:
> Sorting the users by latitude in descending order

If we need a solution that works with a rotated map for notes, shouldn't we try to use the same system here even if we don't need rotation here?
The main map PR will be large enough anyway.

aNsHuL5217 commented:
I’ve updated the code to handle the Z-ordering correctly.

To solve the sorting problem, I added a listener that updates the `zIndex` based on the marker's Y-position on the screen whenever the map moves or rotates. This keeps the "closest" markers on top without breaking the visuals.

Here is a quick video showing the rotation behavior:

https://github.com/user-attachments/assets/c0b4c048-3c75-4ec8-bfdd-5f877eb41d7e

hlfan commented on app/assets/javascripts/dashboard.js:
```diff
@@ -33,14 +33,40 @@ $(function () {
+        const point = map.project([item.lon, item.lat]);
+        const zIndex = Math.round(point.y);
+        item.marker.getElement().style.zIndex = zIndex;
+      });
```
I don't really like that this assigns values in the hundreds for a few dozen markers.
Some method of ending up with a more sequential z-index list would be great.

aNsHuL5217 replied:
That makes sense. I updated the logic to sort the markers by Y-position first, then just assigned a simple index (0, 1, 2...) based on that order.

aNsHuL5217 commented on app/assets/javascripts/dashboard.js:
```diff
@@ -33,14 +33,40 @@ $(function () {
 
+    const markerObjects = [];
+
     $("[data-user]").each(function () {
```
Done. I switched the loop over to a `.map(...).get()` chain. It's definitely cleaner to build the array in one pass rather than pushing to a list manually.

1ec5 commented on app/assets/javascripts/dashboard.js:
```diff
@@ -33,14 +33,40 @@ $(function () {
+        item.currentY = map.project([item.lon, item.lat]).y;
+      });
+      markerObjects.sort((a, b) => a.currentY - b.currentY);
+      markerObjects.forEach((item, index) => {
```
Nit: It looks like other files are using for-of loops instead of `forEach`. Maybe we should be consistent for readability.

hlfan reviewed (approved):
Thanks for your work on this.
